frontend/projects/dipa-frontend/src/app/modules/gantt/gantt.component.ts

```diff
@@ -0,0 +1,64 @@
+import {AfterContentInit, Component, EventEmitter, OnInit, Output, ViewChild} from '@angular/core';
+import {GanttControlsService} from './gantt-controls.service';
+import {ChartComponent} from './chart/chart.component';
+import {TimelineService} from './services/timeline.service';
+
+@Component({
+  selector: 'app-gantt',
+  templateUrl: './gantt.component.html',
+  styleUrls: ['./gantt.component.scss']
+})
+export class GanttComponent implements OnInit, AfterContentInit {
+
+  @Output() dateChange: EventEmitter<any> = new EventEmitter();
+
+  @ViewChild('ganttChart', { static: true }) chart: ChartComponent;
+
+  chartData = [];
+
+  milestoneData = [];
+  taskData = [];
+
+  periodStartDate = new Date(2020, 0, 1);
+  periodEndDate = new Date(2020, 11, 31);
+
+  constructor(public ganttControlsService: GanttControlsService,
+              private timelineService: TimelineService) {
+
+    this.timelineService.getTaskData()
+    .subscribe((data) => {
+      this.taskData = data;
+    });
+
+    this.timelineService.getMilestoneTaskData()
+    .subscribe((data) => {
+      this.milestoneData = data;
+    });
+  }
+
+  ngOnInit(): void {
+    this.dateChange.emit();
+  }
```

Comment:
```suggestion
  constructor(public ganttControlsService: GanttControlsService,
              private timelineService: TimelineService) {  }

  ngOnInit(): void {
    this.timelineService.getTaskData()
    .subscribe((data) => {
      this.taskData = data;
    });

    this.timelineService.getMilestoneTaskData()
    .subscribe((data) => {
      this.milestoneData = data;
    });
    
    this.dateChange.emit();
  }
```
@Bock4Soft Subscriben im Contructor ist ein Anti-Pattern (siehe https://angular.io/tutorial/toh-pt4#call-it-in-ngoninit ) Eigentlich müsste hier noch ein Unsubscribe erfolgen. Da es sich hier um singleton services handelt ist es nicht zwangsläufig nicht nötig. Schön ist aber anders. Hier mal folgenden Artikel https://medium.com/angular-in-depth/the-best-way-to-unsubscribe-rxjs-observable-in-the-angular-applications-d8f9aa42f6a0 lesen. Insbesondere Punkt 5 Async Pipe. Netter Nebeneffekt, diese kann automatisch einen Loading Screen anzeigen.